### ISSUE TITLE
Shapely uses GEOS in /usr/local/lib even when built against a specific version

### ISSUE THREAD

**Issue by tomplex** [Toblerity/Shapely#872]
## Expected behavior and actual behavior.

When you specify a `GEOS_CONFIG` when building Shapely, it should build against and default to the version of GEOS which you specify. Instead when you load Shapely, it defaults to a GEOS which is in `/usr/local/lib`.

## Steps to reproduce the problem.

You'll need multiple versions of GEOS on your machine, one of which is linked into `/usr/local/lib`.

```bash
$ python3 -m venv /tmp/shapely-geos-test
$ source /tmp/shapely-geos-test/bin/activate
$ locate geos-config
/usr/local/Cellar/geos/3.6.3/bin/geos-config
/usr/local/Cellar/geos/3.7.1_1/bin/geos-config
/usr/local/bin/geos-config
# Check the default version of GEOS
$ /usr/local/bin/geos-config --version
3.7.1
$ export GEOS_CONFIG=/usr/local/Cellar/geos/3.6.3/bin/geos-config
$ $GEOS_CONFIG --version
3.6.3
$ git clone git@github.com:Toblerity/Shapely.git /tmp/shapely
$ pip install /tmp/shapely
$ python
```
```python
>>> import logging
>>> logging.basicConfig(level=logging.DEBUG)
>>> from shapely import geos
DEBUG:shapely.geos:Trying `CDLL(/usr/local/lib/libgeos_c.dylib)`
DEBUG:shapely.geos:Library path: '/usr/local/lib/libgeos_c.dylib'
DEBUG:shapely.geos:DLL: <CDLL '/usr/local/lib/libgeos_c.dylib', handle 7fec6f53b630 at 0x109e6ee80>
DEBUG:shapely.geos:Trying `CDLL(/usr/lib/libc.dylib)`
DEBUG:shapely.geos:Library path: '/usr/lib/libc.dylib'
DEBUG:shapely.geos:DLL: <CDLL '/usr/lib/libc.dylib', handle 113c3ff60 at 0x109e6ee50>
>>> geos.geos_version
(3, 7, 1)
```

## Operating system

Mac OS X 10.14.5. I also believe this issue is occurring in the Travis CI builds.

## Shapely version and provenance

Current git master.

Source without stdout:

```bash
# install.sh

python3 -m venv /tmp/shapely-geos-test
source /tmp/shapely-geos-test/bin/activate
locate geos-config
/usr/local/bin/geos-config --version
# You'll need to change this
export GEOS_CONFIG=""
git clone git@github.com:Toblerity/Shapely.git /tmp/shapely
pip install /tmp/shapely
```

```python
# shapely_version.py
import logging
logging.basicConfig(level=logging.DEBUG)

from shapely import geos
print(geos.geos_version)
```

**Comment by tomplex** [comment]
The documentation suggests that if you're building Shapely from source you'll want [cython and numpy](https://github.com/Toblerity/Shapely#source-distributions) installed first, which results in the same behavior as noted here: Shapely links against the GEOS in `/usr/local/lib` when you run it, even though the `speedups` package should have been built against the specified GEOS version.

**Comment by tomplex** [comment]
Seems that Shapely will not install, even if GEOS_CONFIG is specified, unless there is an installation of GEOS in a standard location:

```docker
# shapely.Dockerfile
FROM ubuntu:18.04

RUN apt update && apt install -y \
    wget \
    build-essential \
    cmake \
    git \
    python3 \
    python3-pip \
    locate

ENV GEOSVERSION=3.6.3
ENV GEOS_INSTALL_LOC=/opt/geos/$GEOSVERSION
ENV GEOS_CONFIG=$GEOS_INSTALL_LOC/bin/geos-config

WORKDIR /tmp

RUN wget -q http://download.osgeo.org/geos/geos-$GEOSVERSION.tar.bz2 && \
    tar xfj geos-$GEOSVERSION.tar.bz2 && \
    cd geos-$GEOSVERSION && \
    mkdir build && cd build && \
    cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$GEOS_INSTALL_LOC .. && \
    make -j 4 && \
    make install && \
    rm -rf /tmp/geos-$GEOSVERSION

RUN git clone https://github.com/Toblerity/Shapely.git /tmp/shapely
RUN updatedb

ENTRYPOINT ["/bin/bash"]
```

```bash
$ docker build -t shapely-test -f shapely.Dockerfile .
$ docker run -it shapely-test

root@...:/tmp# $GEOS_CONFIG --version
3.6.3
root@...:/tmp# pip3 install /tmp/shapely
Processing ./shapely
    Complete output from command python setup.py egg_info:
    cannot open shared object from '/opt/geos/3.6.3/bin/geos-config --clibs': '-L/opt/geos/3.6.3/lib -lgeos_c'
    there were 2 guess(es) for this path:
    	/opt/geos/3.6.3/lib/libgeos_c.so
    	/opt/geos/3.6.3/lib/libgeos_c.so.1
    Failed `CDLL(libgeos_c.so.1)`
    Failed `CDLL(libgeos_c.so)`
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-o0ueeaia-build/setup.py", line 85, in <module>
        from shapely._buildcfg import geos_version_string, geos_version, \
      File "/tmp/pip-o0ueeaia-build/shapely/_buildcfg.py", line 170, in <module>
        fallbacks=['libgeos_c.so.1', 'libgeos_c.so'])
      File "/tmp/pip-o0ueeaia-build/shapely/_buildcfg.py", line 164, in load_dll
        libname, fallbacks or []))
    OSError: Could not find library geos_c or load any of its variants ['libgeos_c.so.1', 'libgeos_c.so']

    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-o0ueeaia-build/
root@...:/tmp# locate libgeos_c.so.1
/opt/geos/3.6.3/lib/libgeos_c.so.1
/opt/geos/3.6.3/lib/libgeos_c.so.1.10.3
```

**Comment by sgillies** [comment]
Resolved by #873.
